src/rules/no-multiple-h1.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview Rule to enforce at most one h1 heading in Markdown.
+ * @author Pixel998
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: [{ frontmatterTitle?: string; }]; }>}
+ * NoMultipleH1RuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoMultipleH1RuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description: "Disallow multiple h1 headings in the same document",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-multiple-h1.md",
+		},
+
+		messages: {
+			multipleH1: "More than one h1 heading found.",
```

Comment:
how about this?

```suggestion
			multipleH1: "Only one top-level heading is allowed per document.",
```

Reply:
How about "Unexpected additional h1 heading found."?